lib/Sema/DerivedConformanceParameterAggregate.cpp

```diff
@@ -73,6 +73,27 @@ bool DerivedConformance::canDeriveParameterAggregate(TypeChecker &TC,
   return bool(deriveParameterAggregate_Parameter(TC, nominal));
 }
 
+// Add @_fixed_layout attribute to type conforming to `ParameterAggregate`, if
+// necessary. @_fixed_layout is currently necessary for differentiation wrt
+// structs conforming to `ParameterAggregate` for resilience reasons.
```

Comment:
Please drop the specific reason regarding differentiation in comments. Adding `@_fixed_layout` is a strict requirement for correctness reasons, not just differentiation. If we ever allowed retroactive addition of stored properties, then the semantics of `update` will be incorrect because it won't visit retroactively added stored properties.

Reply:
Thanks for the explanation, that makes sense. Doc comment edited in https://github.com/apple/swift/pull/18507/commits/aa37d55cd2ef66a0a9e8f92278aa62c8406fbe08.